File: Backend/tool_calling.py
# backend/tool_calling.py
from nlp.model import get_intent

def check_eligibility():
    return "You are eligible for a loan based on basic criteria."

def calculate_emi():
    return "Your estimated EMI is ₹10,000 per month."

def tool_calling(user_message):
    intent_data = get_intent(user_message)
    intent = intent_data["intent"]

    if intent == "loan_eligibility":
        return check_eligibility()
    elif intent == "emi_calculation":
        return calculate_emi()
    else:
        return "Sorry, I didn't understand that."

File: Backend/tool_calling.js
// Import the get_intent function (assuming it's in the same directory)
const { getIntent } = require('../nlp/model');

// Function to check eligibility
function checkEligibility() {
    return "You are eligible for a loan based on basic criteria.";
}

// Function to calculate EMI
function calculateEMI() {
    return "Your estimated EMI is ₹10,000 per month.";
}

// Function to handle tool calling based on user message
function toolCalling(userMessage) {
    const intentData = getIntent(userMessage);
    const intent = intentData.intent;

    if (intent === "loan_eligibility") {
        return checkEligibility();
    } else if (intent === "emi_calculation") {
        return calculateEMI();
    } else {
        return "Sorry, I didn't understand that.";
    }
}

// Export the toolCalling function for use in other files
module.exports = { toolCalling };
